File: Algorithms/C++/476-Number_Complement.cpp
// Runtime: 4 ms, faster than 52.83% of C++ online submissions for Number Complement.
// Memory Usage: 8.1 MB, less than 100.00% of C++ online submissions for Number Complement.
class Solution {
public:
    int findComplement(int num) {
        unsigned int mask = ~0;
        while (num & mask)
            mask <<= 1;
        return ~num & ~mask;
    }
};

File: Algorithms/Python3.x/476-Number_Complement.py
# Runtime: 28 ms, faster than 98.42% of Python3 online submissions for Number Complement.
# Memory Usage: 12.6 MB, less than 100.00% of Python3 online submissions for Number Complement.class Solution:
class Solution:
    def findComplement(self, num: int) -> int:
        num = [('1', '0')[int(x)] for x in bin(num)[2:]]
        return int(''.join(num), 2)
